### ISSUE TITLE
VS crashes when I paste this into a C# file

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#9721]
**Version Used**: Visual Studio 2015 24720.00 (Update 1)

**Steps to Reproduce**:

1. Copy and paste [the content of this file](https://gist.github.com/AArnott/cb65e7e7718350d268e5) into an empty .cs document open in VS2015.
2. Select [lines 6-13](https://gist.github.com/AArnott/cb65e7e7718350d268e5#file-c-roslyncrashbug-cs-L6-L13) and copy to clipboard.
3. Select [line 19](https://gist.github.com/AArnott/cb65e7e7718350d268e5#file-c-roslyncrashbug-cs-L19) and Paste (thereby replacing this line with the lines copied in the previous step).

**Expected Behavior**:
A block of text is inserted into a C# method (ready for me to fix up to make syntactically correct).

**Actual Behavior**:
VS crashes (for a loose file as described here), or simply suddenly vanishes when [in the context of the full project](https://github.com/AArnott/PCLCrypto/tree/repros/Roslyn9721).

**Comment by tannergooding** [comment]
FYI. This does not appear to repro in the Update 2 RC

**Comment by Pilchie** [comment]
Yep, but I can repro on U1. Looking to see what the issue was.

**Comment by Pilchie** [comment]
This appears to be an issue in the lexer.  Stack in U1:

```
   at System.Number.ParseInt32(String s, NumberStyles style, NumberFormatInfo info)
   at Microsoft.CodeAnalysis.RealParser.DecimalFloatingPointString.FromSource(String source)
   at Microsoft.CodeAnalysis.RealParser.TryParseDouble(String s, Double& d)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.GetValueDouble(String text)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.ScanNumericLiteral(TokenInfo& info)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.ScanSyntaxToken(TokenInfo& info)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.LexSyntaxToken()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer.Lex(LexerMode mode)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.Reader.LexNewToken(LexerMode mode)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.Reader.ReadNewToken(LexerMode mode)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Blender.Reader.ReadNodeOrToken(LexerMode mode, Boolean asToken)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxParser.AddNewToken()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.SyntaxParser.FetchCurrentToken()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParsePostFixExpression(ExpressionSyntax expr)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseTerm(UInt32 precedence)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseSubExpressionCore(UInt32 precedence)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseSubExpression(UInt32 precedence)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseObjectOrCollectionInitializerMember(Boolean& isObjectInitializer)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseObjectOrCollectionInitializerMembers(SyntaxToken& startToken, SeparatedSyntaxListBuilder`1 list, Boolean& isObjectInitializer)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseObjectOrCollectionInitializer()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseArrayOrObjectCreationExpression()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseNewExpression()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseTerm(UInt32 precedence)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseSubExpressionCore(UInt32 precedence)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseSubExpression(UInt32 precedence)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseReturnStatement()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseStatementNoDeclaration(Boolean allowAnyExpression)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseStatementCore()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseStatements(CSharpSyntaxNode& previousNode, SyntaxListBuilder`1 statements, Boolean stopOnSwitchSections)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseBlock(Boolean isMethodBody, Boolean isAccessorBody)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseBlockAndExpressionBodiesWithSemicolon(BlockSyntax& blockBody, ArrowExpressionClauseSyntax& expressionBody, SyntaxToken& semicolon)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseMethodDeclaration(SyntaxListBuilder`1 attributes, SyntaxListBuilder modifiers, TypeSyntax type, ExplicitInterfaceSpecifierSyntax explicitInterfaceOpt, SyntaxToken identifier, TypeParameterListSyntax typeParameterList)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseMemberDeclarationOrStatement(SyntaxKind parentKind, String typeName)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseClassOrStructOrInterfaceDeclaration(SyntaxListBuilder`1 attributes, SyntaxListBuilder modifiers)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseTypeDeclaration(SyntaxListBuilder`1 attributes, SyntaxListBuilder modifiers)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseMemberDeclarationOrStatement(SyntaxKind parentKind, String typeName)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseNamespaceBody(SyntaxToken& openBrace, NamespaceBodyBuilder& body, SyntaxListBuilder& initialBadNodes, SyntaxKind parentKind)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseNamespaceDeclaration()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseNamespaceBody(SyntaxToken& openBrace, NamespaceBodyBuilder& body, SyntaxListBuilder& initialBadNodes, SyntaxKind parentKind)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseCompilationUnitCore()
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseWithStackGuard[TNode](Func`1 parseFunc, Func`1 createEmptyNodeFunc)
   at Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.LanguageParser.ParseCompilationUnit()
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTree.WithChanges(SourceText newText, IReadOnlyList`1 changes)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTree.WithChangedText(SourceText newText)
   at Microsoft.CodeAnalysis.DocumentState.<IncrementallyParseTreeAsync>d__13.MoveNext()
```

**Comment by khyperia** [comment]
I believe this was fixed due to #6844 (similar issue: #8480 ). In particular, the "DP" line in the gist contains the string "81e352855994081499", enough numbers in the "exponent" to trigger the overflow/crash.

**Comment by gafter** [comment]
Dup of #8480
